### PR TITLE
Convert int8 numpy array to CharTensor

### PR DESCRIPTION
**Pull request by ssnl**
When rewriting `default_collate`, I noticed that `from_numpy` and `as_tensor` and `tensor` all do not work on `np.int8` arrays.

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by SsnL** [comment]
@soumith could you reland? I pushed a new commit.

**Comment by soumith** [comment]
sure!

File: torch/csrc/utils/tensor_numpy.cpp [hunk 1]
```diff
@@ -166,6 +166,7 @@ ScalarType numpy_dtype_to_aten(int dtype) {
     case NPY_HALF: return kHalf;
     case NPY_INT32: return kInt;
     case NPY_INT16: return kShort;
+    case NPY_INT8: return kChar;
```
Comment (fmassa) [inline torch/csrc/utils/tensor_numpy.cpp]: question: is `char` signed by default for all architectures?

Comment (fmassa) [reply torch/csrc/utils/tensor_numpy.cpp]: Disregard my question, we have modified the code to [use `int8_t` already](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Dispatch.h#L59)

**Pull request closed without merging**